rails/app/models/exercise_entry.rb

```diff
@@ -4,4 +4,6 @@ class ExerciseEntry < ApplicationRecord
   belongs_to :user
 
   encrypts :name
+
+  scope :recent, -> { distinct(:name).order(created_at: :desc).limit(3) }
```

Comment:
You have this scope three times. Maybe you can extract it into a concern? `included do scope :recent...`

Concern name could be `Recent`.